parser-typechecker/src/Unison/Codebase/Branch.hs

```diff
@@ -475,18 +499,22 @@ empty0 :: Branch0 m
 empty0 =
   Branch0 mempty mempty mempty mempty mempty mempty mempty mempty mempty mempty
 
+-- | Checks whether a Branch0 is empty.
 isEmpty0 :: Branch0 m -> Bool
 isEmpty0 = (== empty0)
 
+-- | Checks whether a branch is empty AND has no history.
 isEmpty :: Branch m -> Bool
 isEmpty = (== empty)
 
+-- | Perform an update over the current branch and create a new causal step.
+-- Note: this does not step history on updated child branches, see 'stepManyAt'.
```

Comment:
The note is probably not relevant since all of the blessed ways of updating child branches will handle stepping properly anyways, I'll just remove the note.